### PR TITLE
[ivalue] operator<<: don't error on invalid IValue tags

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112232

While running the profiler, we observed a scenario where we observe IValues with invalid tags. Specifically, we try to convert the IValue to a string here:

https://github.com/pytorch/pytorch/blob/d3bf6803b62c79f1dafd1eec49b4bd65d5a27697/torch/csrc/profiler/util.cpp#L306-L308

and in the scenario with invalid IValues, an exception gets thrown here, in `operator<<`.

https://github.com/pytorch/pytorch/blob/d3bf6803b62c79f1dafd1eec49b4bd65d5a27697/aten/src/ATen/core/ivalue.cpp#L864

IMO, `<<` shouldn't error if the IValue is bad; instead, we should just print that the IValue tag is invalid.

Differential Revision: [D50760040](https://our.internmc.facebook.com/intern/diff/D50760040)

### CODE REVIEW

File: aten/src/ATen/core/ivalue.cpp [hunk 1]
```diff
@@ -861,7 +861,7 @@ std::ostream& operator<<(std::ostream & out, const IValue & v) {
     }
 
   }
-  AT_ERROR("Tag not found: ", v.tagKind());
+  return out << "InvalidIValue(tag=" << std::to_string(static_cast<uint32_t>(v.tag)) << ")";
```
Comment (albanD) [inline aten/src/ATen/core/ivalue.cpp:864]: nit: this is not consistent with other objects being printed here. But I guess that's fine.

**Review by albanD — APPROVED** [review]

File: aten/src/ATen/core/ivalue.cpp (hunk 1 above)
Comment (davidberard98) [reply aten/src/ATen/core/ivalue.cpp:864]: good point - switched it to `<Invalid IValue tag=1234>`

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50760040).

**Pull request closed without merging**